### DISCUSSION TITLE
Corfu child frame gets stuck

### DISCUSSION THREAD

**Discussion by guibor** [minad/corfu#453] [Q&A]
When I type, corfu completion popup works as designed.

I'm probably not the only case (I found [this](https://www.reddit.com/r/emacs/comments/18ikiek/corfus_posframe_sometimes_gets_stuck_how_can_i/) thread on reddit as well)

But very often, it just gets stuck. The window just gets stuck there and leaves an empty grey rectangle. The only way to get it off, is to trigger corfu again or call completion at point.

I use mac OS X with the following setup:

```
(use-package corfu
    :custom
    (corfu-cycle t)
    (corfu-auto t)
    (corfu-auto-delay 2)
    (corfu-auto-prefix 4)
    :init
    (global-corfu-mode)
 
    (defun corfu-enable-in-minibuffer ()
      "Enable Corfu in the minibuffer."
      (when (local-variable-p 'completion-at-point-functions)
        (setq-local corfu-echo-delay nil ;; Disable automatic echo and popup
                    corfu-popupinfo-delay nil)
        (corfu-mode 1)))
    (add-hook 'minibuffer-setup-hook #'corfu-enable-in-minibuffer))
    ```
    
    And thanks a lot for the great work on this great package!

**Comment by minad** [comment]
Hi! Can you provide a minimal recipe based on emacs -Q, such that I can also observe this? See https://github.com/minad/consult?tab=readme-ov-file#bug-reports for an exemplary list of all the needed information. It could be that the bug is Mac-specific, and as such must be fixed in the Mac port of Emacs. However many people happily use Corfu on Mac, so maybe there is some subtlety in your setup.

**Comment by guibor** [comment]
Unfortunately it is hard to reproduce consistently, it happens every now and then (but when it does, it is very annoying).
It seems to be related to clearing the posframe, perhaps you can suggest a direction worth investigating?

**Comment by Nebucatnetzer** [comment]
I see this as well.
It happens only in modes where I use lsp, I think.
At least I haven't seen in Elisp-Mode yet.

**Comment by jdtsmith** [comment]
I do also _very occasionally_ see this.  I suspect it has to do with the frame focus being taken from the child-frame due to some timing specific to MacOS not found elsewhere.  Next time you see if it occurs, check if you've done something recently to change frames.

**Comment by saffronmciver** [comment]
I also see this but on tty on linux. For me, I think the issue was `select-frame` being called on `corfu--frame` making it visible. Calling `(make-frame-invisible corfu--frame)` removed it. This happened due to packages calling `(with-selected-frame)` on all live frames making `corfu--frame` visible.

**Reply by minad** [reply]
This has been fixed upstream in Emacs.
